test/compilable/test17954.d

```diff
@@ -0,0 +1,38 @@
+// https://issues.dlang.org/show_bug.cgi?id=17954
+/*
+TEST_OUTPUT:
+---
+compilable/test17954.d(32): Deprecation: declaring a member named `init` is deprecated
```

Comment:
message can be improved, it now says "deprecation" twice.
Proposal: "Deprecation: use of reserved name `init` as member name"